subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/modulecache/ModuleMetadataSerializer.java

```diff
@@ -300,7 +301,7 @@ private void writeMavenDependency(MavenDependencyDescriptor mavenDependency) thr
             writeNullableArtifact(mavenDependency.getDependencyArtifact());
             writeMavenExcludeRules(mavenDependency.getAllExcludes());
             encoder.writeSmallInt(mavenDependency.getScope().ordinal());
-            encoder.writeBoolean(mavenDependency.isOptional());
+            encoder.writeSmallInt(mavenDependency.getType().ordinal());
```

Comment:
Since we use `ordinal` here it would be good to add a note to the docs of the enum to warn that we shouldn't change the order of values, or it would break the binary format.

Reply:
Done, for `MavenScope` as well.